test/pylib/scylla_cluster.py

```diff
@@ -419,12 +426,16 @@ async def start(self, api: ScyllaRESTAPIClient) -> None:
             if hasattr(self, "host_id") or await self.get_host_id(api):
                 cql_up_state = await self.cql_is_up()
                 if cql_up_state == CqlUpState.QUERIED:
+                    if expected_error is not None:
+                        raise RuntimeError(f"Server {self.server_id}, IP = {self.ip_addr}, started but was expected to fail with error [{expected_error}]")
                     return
 
             # Sleep and retry
             await asyncio.sleep(sleep_interval)
 
         err = f"Failed to start server with ID = {self.server_id}, IP = {self.ip_addr}."
+        if expected_error is not None:
+            err += f" Expected error [{expected_error}]"
```

Comment:
This interrupts the "flow" of the message, which says: "failed to start", and then gives the specifics of the failure (i.e. managed to obtain host ID and so on).

Let's put this on a new line, before "Check server log at ..." to not break the flow. Also provide more details: where was this error expected? (In the log file of the server, and not e.g. during a CQL query.)

It would be also great to include information whether the message appeared in the log. You do include it in case the process stops, but not in case of timeout which this code is handling.